### PR TITLE
Hex logo

### LINKED ISSUE #165: Add a hex logo

### PR DESCRIPTION
**Pull request by villegar**
- [x] added unit tests and checked code coverage with `covr::report()` (should aim for 100%)
- [x] ran `devtools::document()`
- [x] ran `lintr::lint_package()` and resolved all lint warnings and notes
- [x] ran `styler::style_pkg()` to make sure code matches the style guidelines
- [x] ran R-CMD CHECK and resolved all issues

This pull request closes #165

### CODE REVIEW

**Comment by tomjemmett** [comment]
This logo looks amazing! Thanks! On mobile so will properly review later, quick suggestion is we move the script for generating the logo to be a dev dependency in the inst/ folder, and move the hexSticker depends to suggests, but otherwise looks great

File: R/hex-logo.R [hunk 1]
```diff
@@ -0,0 +1,157 @@
+#' Generate hexagonal logo for the package
```
Comment (tomjemmett) [inline R/hex-logo.R:1]: I would remove this as a function that is part of the package, but include as `inst/hex-logo.R`. It's not something that should be run other than by developers to generate the logo initially.

File: README.Rmd [hunk 2]
```diff
@@ -15,7 +15,8 @@ knitr::opts_chunk$set(
 )
 ```
 
-# NHSRplotthedots <a alt="NHS-R Community's logo" href='https://nhsrcommunity.com/'><img src='https://nhs-r-community.github.io/assets/logo/nhsr-logo.png' align="right" height="80" /></a>
+# NHSRplotthedots <img src='inst/images/logo.png' align='right' height='200' />
```
Comment (tomjemmett) [inline README.Rmd:18]: thoughts from @Lextuga007 / @ChrisBeeley welcome here... most of our repos use the NHS-R logo on the README, should we continue using this approach with this repo, or use the hex sticker instead?

File: README.Rmd [hunk 3]
```diff
@@ -15,7 +15,8 @@ knitr::opts_chunk$set(
...
+<!-- <a alt="NHS-R Community's logo" href='https://nhsrcommunity.com/'><img src='https://nhs-r-community.github.io/assets/logo/nhsr-logo.png' align="right" height="80" /></a> -->
```
Comment (tomjemmett) [inline README.Rmd:19]: see previous comment, but remove commented code

**Review by tomjemmett — CHANGES_REQUESTED** [review]
this looks amazing, many thanks! See comments.

One thing that might be worth looking at is the font in the logo, I can't see how that's controlled? Ideally we should probably  follow the [NHS Identity](https://www.england.nhs.uk/nhsidentity/identity-guidelines/fonts/) for fonts, but the primary font is a commercial font, with the fallback being Arial (non open license, but it is included in Windows)

**Comment by ThomUK** [comment]
It's great that you've turned this around so quickly @villegar!  I can't see the logo here because I'm on mobile but just in case...

Is there any chance of merging the design with the one from Deb shared by ChrisB? The particular parts of that design that are strong are:
* Designed by a non-r user, showing collaboration without code.
* Championed by the MDC team, who inspired the package.
* The dot colours and line colour are recognisable and 'on brand'

I agree with @Lextuga007 too that NHSRplothedots should be on the hex so that the package name is present.

Apols if all this is already picked up! Just my 2p without being able to see the latest design! Great work!

File: README.Rmd (hunk 2 above)
Comment (villegar) [reply README.Rmd:18]: This sounds like a potential "NHS-Rverse" 🤩

File: R/hex-logo.R (hunk 1 above)
Comment (villegar) [reply R/hex-logo.R]: I have now moved the file to `inst` and added two new parameters: `p_family` and `p_fontface`, which can be used to control the font of the package name and URL.

**Comment by villegar** [comment]
@ThomUK and @Lextuga007 thanks for the feedback! I think it should be possible to do some "mash-up" and combine all the ideas into a final logo. I will try to recreate Deb's design using `ggplot2` (at least the main plot should be easy) and then generate a sticker with `hexSticker`.

**Comment by villegar** [comment]
My attempt to recreate Deb's hex code with `ggplot2`, we need some final tweaks, but it's almost there:

**Comment by ThomUK** [comment]
@villegar this is amazing! Sorry I didn't see it sooner (Zoë just posted it on twitter which is how I saw it). Great work!

**Comment by Lextuga007** [comment]
Deb Cundey made some great points (via Twitter) which I've added my views to:

* the NHS logo must be official and that extends to fonts and colours

A great spot from Deb and my subsequent view is that we don't use the official NHS logo as that will require, I think, confirmation from someone, somewhere in the NHS. Just stating it is used by the NHS in uncoloured font could suffice? 

Given that this package was not funded by the NHS, Making Data Count, NHS England or NHS-R Community then it doesn't have to carry any specific logos. Whilst using the NHS logo gives credence to the package, I think we'll need more time to get that agreed. We could use the NHS-R Community logo as this was produced by Community members.

Open to discussion on this!

* Deb likes the NHS-R Community GitHub link wording but aesthetically thinks it would look nicer parallel to one of the hex sides

Personal views on this are that we don't include it at all and try to keep the image as simple as possible. Things like repo links could, possibly, get changed and that would date the logo. Saying that, removing it from that side may unbalance the image - I'm not sure.

**Review by tomjemmett — CHANGES_REQUESTED** [review]
This looks good to me now, suggest we go with just the one file and call it inst/images/logo.png, and likewise just have the one generate script called inst/hex-logo.R

**Comment by villegar** [comment]
> This looks good to me now, suggest we go with just the one file and call it inst/images/logo.png, and likewise just have the one generate script called inst/hex-logo.R

@tomjemmett I have removed the old versions of the logo and the additional script. Let me know if there are any additional changes needed.

**Review by tomjemmett — APPROVED** [review]

**Pull request merged**